irohad/main/impl/consensus_init.cpp

```diff
@@ -94,8 +95,14 @@ namespace iroha {
               async_call,
           std::shared_ptr<shared_model::interface::CommonObjectsFactory>
               common_objects_factory) {
+        std::shared_ptr<iroha::consensus::yac::CleanupStrategy>
```

Comment:
Won't `auto` work?

Reply:
It could be, but types on left and right sides are different.